tests/test_skyWcs.py

```diff
@@ -0,0 +1,123 @@
+
+        Returns
+        -------
+        wcs : SkyWcs
+            The generated pure TAN SkyWcs
+        """
+        wcs = SkyWcs(
+            crpix = self.crpix,
+            crval = self.crval,
+            cdMatrix = makeCdMatrix(scale=self.scale, orientation=orientation, flipX=flipX),
+        )
+        self.checkPersistence(wcs)
+
+        xoffAng = 180*degrees if flipX else 0*degrees
+
+        pixelList = [
+            Point2D(self.crpix[0], self.crpix[1]),
+            Point2D(self.crpix[0] + 1, self.crpix[1]),
+            Point2D(self.crpix[0], self.crpix[1] + 1),
+        ]
+        skyList = wcs.tranForward(pixelList)
+
+        # check pixels to sky
+        predSkyList = [
+            self.crval,
+            self.crval.offset(xoffAng - orientation, self.scale),
+            self.crval.offset(90*degrees - orientation, self.scale)
+        ]
+        self.assertSpherePointListsAlmostEqual(predSkyList, skyList)
+        self.assertSpherePointListsAlmostEqual(predSkyList, wcs.pixelToSky(pixelList))
+        for pixel, predSky in zip(pixelList, predSkyList):
+            self.assertSpherePointsAlmostEqual(predSky, wcs.pixelToSky(pixel))
+            anglePair = wcs.pixelToSky(*pixel)
+            self.assertSpherePointsAlmostEqual(predSky, SpherePoint(*anglePair))
+
+        # check sky to pixels
+        self.assertPairListsAlmostEqual(pixelList, wcs.tranInverse(skyList))
+        self.assertPairListsAlmostEqual(pixelList, wcs.tranInverse(skyList))
+        for pixel, sky in zip(pixelList, skyList):
+            self.assertPairsAlmostEqual(pixel, wcs.skyToPixel(sky))
+            xyPair = wcs.skyToPixel(*sky)
+            self.assertPairsAlmostEqual(pixel, Point2D(*xyPair))
+
+        crval = wcs.getSkyOrigin()
+        self.assertSpherePointsAlmostEqual(crval, self.crval)
+
+        crpix = wcs.getPixelOrigin()
+        self.assertPairsAlmostEqual(crpix, self.crpix)
+
+        cdMatrix = wcs.getCdMatrix()
+        predCdMatrix = makeCdMatrix(scale=self.scale, orientation=orientation, flipX=flipX)
+        self.assertFloatsAlmostEqual(cdMatrix, predCdMatrix)
+
+        pixelScale = wcs.getPixelScale(self.crpix)
+        self.assertAnglesAlmostEqual(self.scale, pixelScale)
+
+        # Compute a WCS with the pixel origin shifted by an arbitrary amount
+        # The resulting sky origin should not change
+        offset = Extent2D(500, -322)  # arbitrary
+        shiftedWcs = wcs.shiftedPixelOrigin(*offset)
+        predShiftedPixelOrigin = self.crpix + offset
+        self.assertPairsAlmostEqual(shiftedWcs.getPixelOrigin(), predShiftedPixelOrigin)
+        self.assertSpherePointsAlmostEqual(shiftedWcs.getSkyOrigin(), self.crval)
+
+        shiftedPixelList = [p + offset for p in pixelList]
+        shiftedSkyList = shiftedWcs.tranForward(shiftedPixelList)
+        self.assertSpherePointListsAlmostEqual(skyList, shiftedSkyList)
+
+        return wcs
+
+    def testOrient0FlipXFalse(self):
```

Comment:
I don't think a TanWcs with CRVAL at the north and south poles is very well defined in the sense that there's no way to specify the position angle (which way is "N and E" the CCD). I'm going to leave off that test and just test very near the pole for now. If you think it important I will gladly look into this again when I understand better what to expect at the poles. Note: I realize it is possible to map X,Y <-> RA/Dec  even with CRVAL at a pole (and Wcs and SkyWcs will presumably both do so without complaining); I'm just saying that I think the code has to make an arbitrary decision to do so, which makes it harder to test.